lib/deploy.js

```diff
@@ -32,7 +32,20 @@ module.exports = promisify(function(config, callback) {
   var ghPagesConfig = {};
   var rootDir = config.rootDir ? config.rootDir : '.';
   var cloneDir = config.cloneDir ? config.cloneDir : null;
-  var commitMessage = config.message ? config.message : childProcess.execSync('git log --format=%B -n 1').toString().trim();
+
+  var commitMessage = config.message || (function() {
+    var spawn = childProcess.spawnSync('git', ['log', '--format=%B', '-n', '1']);
+    var errorText = spawn.stderr.toString().trim();
+
+    if (errorText) {
+      gutil.log('Fatal error from `git log`.  You must have one commit before deploying.');
+      throw new Error(errorText);
+      callback(spawn.stderr);
```

Comment:
Ah, I didn't notice this was called inside another function, so the return isn't affecting the enclosing function.

You could use *throw new Error* then and modify the gulp script from:
```JavaScript
gulp.task('deploy', function(callback) {
  oghliner.deploy({
    rootDir: 'dist',
  }, callback);
});
```
to:
```JavaScript
gulp.task('deploy', function() {
  return oghliner.deploy({
    rootDir: 'dist',
  });
});
```
This way the gulp task halts when the error is thrown because the promise returned by deploy gets rejected.

Or you could remove the internal function, write everything in the enclosing function and use `callback(...) return;`.

I'd prefer the second option (or any other option where the callback is called and the gulp task halted) because I think in the future we can avoid using *promisify* and directly return a promise here, so a call to *callback* could be simply replaced by a call to *reject*.